ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -31,9 +31,8 @@ import (
 
 func (o *AviObjectGraph) ProcessL7Routes(key string, routeModel RouteModel, parentNsName string, childVSes map[string]struct{}) {
 	for _, rule := range routeModel.ParseRouteRules().Rules {
-
 		// TODO: add the scenarios where we will not create child VS here.
-		if rule.Matches == nil || rule.Backends == nil {
```

Comment:
it will be good to call `BuildPGPool` only when `rule.Backends !=nil`. This way we will avoid un-necessary assignment and processing.

Reply:
In that case redirect wont work because in redirect's case, `rule.Backends == nil` always

Reply:
In redirect case, we just need to create http policyset. We don't require to buildPG or pools for that. Right?

Reply:
Lets consider a case where we had filters without ReqeustRedirect, as per convention, it will create `pool/poolgroup/backend-svcs`. Now if we update the filters to have RequestRedirect, all the `pools/poolgroups/service` attached earlier must be removed. This piece of code is also taken care of inside `BuildChildVs->BuildPGPool`